state/core.go

```diff
@@ -199,6 +227,71 @@ func (c *Core) Height(commit flow.StateCommitment) (uint64, error) {
 	return height, nil
 }
 
+func (c *Core) Events(height uint64, types ...string) ([]flow.Event, error) {
+	// Make sure that the request is for a height below the currently active
+	// sentinel height; otherwise, we haven't indexed yet and we might return
+	// false information.
+	if height > c.height {
+		return nil, fmt.Errorf("unknown height (current: %d, requested: %d)", c.height, height)
+	}
+
+	lookup := make(map[uint64]struct{})
+	for _, typ := range types {
+		lookup[xxhash.Checksum64([]byte(typ))] = struct{}{}
+	}
+
+	// Iterate over all keys within the events index which are prefixed with the right block height.
+	var events []flow.Event
+	prefix := make([]byte, 1+8)
+	prefix[0] = model.PrefixEventIndex
+	binary.BigEndian.PutUint64(prefix[1:], height)
```

Comment:
I still prefer putting the end index of the slice explicitly, as it makes the code more readable / understandable.